packages/common/src/lib/workspace/project-graph.ts

```diff
@@ -0,0 +1,233 @@
+          source: rootProjectName,
+          target: depProjectName,
+          type: DependencyType.static,
+          sourceFile: joinPathFragments(rootProjectFolder, rootPkgInfo.packageFile),
+        }
+      );
+    }
+  });
+
+  rootPkgInfo.modules?.forEach((childModuleName) => {
+    dependencies.push(
+      {
+        source: rootProjectName,
+        target: childModuleName,
+        type: DependencyType.static,
+        sourceFile: joinPathFragments(rootProjectFolder, rootPkgInfo.packageFile)
+      }
+    );
+  });
+
+  return dependencies;
+}
+
+export function getProjectGraph(
+  pluginName: string,
+  projectFilter: (project: { root: string }) => boolean,
+  getPackageInfo: (project: { root: string }) => PackageInfo,
+  graph: ProjectGraph,
+  context: ProjectGraphProcessorContext
+): ProjectGraph {
+  const builder = new ProjectGraphBuilder(graph);
+  if (process.env['NX_VERBOSE_LOGGING'] === 'true') {
+    logger.debug(
+      `[${pluginName}]: Looking related projects inside the workspace...`
+    );
+  }
+  const workspace = getPackageInfosForNxProjects(
+    pluginName,
+    projectFilter,
+    getPackageInfo,
+    context.projectsConfigurations
+  );
+  Object.entries(workspace.projects).forEach(([projectName, pkgInfo]) => {
+    addDependenciesForProject(
+      pluginName,
+      graph.nodes[projectName].data.root,
+      projectName,
+      pkgInfo,
+      builder,
+      workspace
+    );
+  });
+  return builder.getUpdatedProjectGraph();
+}
+
+//Project Graph V2
+
+export const createNodesFor = (projectFilePattern: string[], pluginName: string) =>
+  [
+    `**/{${projectFilePattern.join(',')}}` as string,
+    (
+      file: string,
+      opt: unknown,
+      context: CreateNodesContext,
+    ): CreateNodesResult => {
+
+      const { root, name } = getNameAndRoot(file);
+
+      return {
+        projects: {
+          [name]: {
+            name,
+            root,
+            tags: [`type:${pluginName.replace('@nxrocks/','')}`],
+          },
+        },
+      };
+    }
+  ] as const;
+
+
+export const createDependenciesIf = (pluginName: string,
+  projectFilePattern: string[],
+  projectFilter: (project: { root: string }) => boolean,
+  getPackageInfo: (project: { root: string }) => PackageInfo,
+  ctx: CreateDependenciesContext) => {
+  if (process.env['NX_VERBOSE_LOGGING'] === 'true') {
+    logger.debug(
+      `[${pluginName}]: Looking related projects inside the workspace...`
+    );
+  }
+  const workspace = getPackageInfosForNxProjects(
+    pluginName,
+    projectFilter,
+    getPackageInfo,
+    { projects: ctx.projects }
+  );
+
+  const dependencies: RawProjectGraphDependency[] = [];
+
+  for (const source in ctx.filesToProcess.projectFileMap) {
+    const changed = ctx.filesToProcess.projectFileMap[source];
+    for (const file of changed) {
+      if (projectFilePattern.some(p => minimatch(file.file, p))) {
+        const { root, name } = getNameAndRoot(file.file);
+
+        dependencies.concat(
```

Comment:
`.concat` doesn't mutate the array, so this call isn't actually doing anything I think?